src/RustyObjectStore.jl

```diff
@@ -81,52 +98,203 @@ function init_object_store(config::ObjectStoreConfig=DEFAULT_CONFIG)
     return nothing
 end
 
+"""
+    $TYPEDEF
+
+# Keyword Arguments
+- `request_timeout_secs::Option{Int}`: (Optional) Client request timeout in seconds.
+- `connect_timeout_secs::Option{Int}`: (Optional) Client connection timeout in seconds.
+- `max_retries::Option{Int}`: (Optional) Maximum number of retry attempts.
+- `retry_timeout_secs::Option{Int}`: (Optional) Maximum amount of time from the initial request after which no further retries will be attempted (in seconds).
+"""
+struct ClientOptions
+    params::Dict{String, String}
+
+    function ClientOptions(;
+        request_timeout_secs::Option{Int} = nothing,
+        connect_timeout_secs::Option{Int} = nothing,
+        max_retries::Option{Int} = nothing,
+        retry_timeout_secs::Option{Int} = nothing
+    )
+        params = Dict()
+        if !isnothing(request_timeout_secs)
+            params["timeout"] = "$(string(request_timeout_secs))s"
```

Comment:
more idiomatic than calling `string` inside string interpolation would be to just use `string` with multiple arguments, like:
```suggestion
            params["timeout"] = string(request_timeout_secs, "s")
```
(same for the other similar cases)

Reply:
Will do!

Reply:
Note that the `string` function has a fast path when inputs share the same type, e.g.:
```
julia> @btime string(string(1), "s")
  27.261 ns (3 allocations: 112 bytes)
"1s"

julia> @btime string(1, "s")
  106.952 ns (5 allocations: 240 bytes)
"1s"
```

Reply:
well that's extremely silly. Why wouldn't Julia just dispatch `string(x, y)` to `string(string(x), string(y))` if that's more performant? shall we go do that in Base?